source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
+
+.. code-block:: none
+   :copyable: false
+
+   _id: 5fb5fad05f734e3794741a35, name: Gabriel García Márquez, lastUpdated: Thu Mar 26 00:00:00 EDT 2020
+   title: One Hundred Years of Solitude, yearPublished: 1967
+   title: Chronicle of a Death Foretold, yearPublished: 1981
+   title: Love in the Time of Cholera, yearPublished: 1985
+
+For more information on retrieving and manipulating MongoDB data, see our
+:doc:`CRUD guide </fundamentals/crud>`.
+
+For more information on this class, see the
+:java-docs:`Document API documentation </apidocs/bson/org/bson/Document.html>`.
 
 BsonDocument
 ------------
 
-- Description
-- Link to Fundamentals > Document Data Formats: BSON for more information
-- Code example of declaring one and appending fields
-- Link to Fundamentals > CRUD for more information
-- Link to API documentation
+The ``BsonDocument`` class provides a type-safe API to access and manipulate
+a BSON document. You need to specify the BSON type from the Java BSON
+library for each field. See the following table for mappings between
+frequently-used BSON and Java BSON library types:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - BSON type
+     - Java BSON library type
+   * - Array
+     - ``org.bson.BsonArray``
+   * - Binary
+     - ``org.bson.BsonBinary``
+   * - Boolean
+     - ``org.bson.Boolean``
+   * - Date (long value)
+     - ``org.bson.BsonDateTime``
+   * - Document
+     - ``org.bson.BsonDocument``
+   * - Double
+     - ``org.bson.BsonDouble``
+   * - Int32
+     - ``org.bson.BsonInt32``
+   * - Int64
+     - ``org.bson.BsonInt64``
+   * - Null
+     - ``null``
+   * - ObjectId
+     - ``org.bson.BsonObjectId``
+   * - String
+     - ``org.bson.BsonString``
+
+
+You can also customize the type mapping by specifying a custom codec.
+For more information on customizing mapped types, see our guide on using
+:doc:`Codecs </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate and build a sample
+``BsonDocument`` instance that contains a variety of data types:
+
+.. code-block:: java
+   :copyable: true
+
+   BsonDocument author = new BsonDocument()
+       .append("_id", new BsonObjectId())
+       .append("name", new BsonString("Gabriel García Márquez"))
+       .append("lastUpdated", new BsonDateTime(new Date(120, 2, 26).getTime()))
+       .append("novels", new BsonArray(Arrays.asList(
+           new BsonDocument().append("title", new BsonString("One Hundred Years of Solitude")).append("yearPublished", new BsonInt32(1967)),
+           new BsonDocument().append("title", new BsonString("Chronicle of a Death Foretold")).append("yearPublished", new BsonInt32(1981)),
+           new BsonDocument().append("title", new BsonString("Love in the Time of Cholera")).append("yearPublished", new BsonInt32(1985))
+      )));
+
+You can insert this document into a collection using the following code:
+
+.. code-block:: java
+   :copyable: true
```

Comment:
While you have specified that these blocks should be copyable, I don't see a copy button in your staged version. Nextgen bug?

Reply:
Will investigate. When I last checked, the first code blocks I created and previewed were working fine, but maybe I introduced some syntax issue along the way